nemo/collections/nlp/models/text_classification/text_classification_model.py

```diff
@@ -112,19 +129,35 @@ def validation_step(self, batch, batch_idx):
 
         val_loss = self.loss(logits=logits, labels=labels)
 
-        # TODO - add eval - callback?
-        # labels_hat = torch.argmax(y_hat, dim=1)
-        # n_correct_pred = torch.sum(y == labels_hat).item()
-        return {'val_loss': val_loss}
+        preds = torch.argmax(logits, axis=-1)
+        tp, fp, fn = self.classification_report(preds, labels)
+
+        tensorboard_logs = {'val_loss': val_loss, 'tp': tp, 'fn': fn, 'fp': fp}
+
+        return {'val_loss': val_loss, 'log': tensorboard_logs}
 
     def validation_epoch_end(self, outputs):
         """
         Called at the end of validation to aggregate outputs.
         :param outputs: list of individual outputs of each validation step.
         """
+        # TODO: Check why need this?
```

Comment:
do we need this?

Reply:
I think @ericharper has added this if to check the outputs. I am not sure why outputs can be none here. I would resolve it later. Put a tag here not to forget it.